pkg/objectstate/rte/machineconfigpool.go

```diff
@@ -21,17 +21,20 @@ import (
 	"fmt"
 
 	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
 	"sigs.k8s.io/controller-runtime/pkg/client"
 
 	machineconfigv1 "github.com/openshift/machine-config-operator/pkg/apis/machineconfiguration.openshift.io/v1"
 
 	nropv1 "github.com/openshift-kni/numaresources-operator/api/v1"
 	nodegroupv1 "github.com/openshift-kni/numaresources-operator/api/v1/helper/nodegroup"
 	"github.com/openshift-kni/numaresources-operator/internal/api/annotations"
+	"github.com/openshift-kni/numaresources-operator/pkg/hash"
 	"github.com/openshift-kni/numaresources-operator/pkg/objectnames"
 	"github.com/openshift-kni/numaresources-operator/pkg/objectstate"
 	"github.com/openshift-kni/numaresources-operator/pkg/objectstate/compare"
 	"github.com/openshift-kni/numaresources-operator/pkg/objectstate/merge"
+	rteupdate "github.com/openshift-kni/numaresources-operator/pkg/objectupdate/rte"
```

Comment:
ok, this is the only remaining item (extra logging is a nice bonus but not required).
Conceptually, `objectstate` sits below in the stack with respect to `objectupdate`:

```
controller
objectupdate
objectstate
client
```
IOW, objectstate can't import objectupdate, but objectupdate can call objectstate (controller can and do import them directly).
We can bypass layers from up to down, but we should never break layering from the bottom to the up

Reply:
And that's why I preferred passing the configmap, because it is more align with current flow. Anyway I'll fix that, I'll find a way to update the ds later in the code

Reply:
I may be wrong but  I think there should be a reasonnably easy way to not break the layering and yet only carry to hash around